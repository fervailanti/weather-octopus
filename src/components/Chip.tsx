import React, { FC, ReactElement } from 'react'
import Title from './Title'

interface ChipProps {
  title: string
  subtitle: string
  image: string
  action?: () => void
}

const Chip: FC<ChipProps> = ({ title, subtitle, image, action }): ReactElement => {
  const transition = 'transition duration-150 ease-in-out'
  const hover = 'lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300'
  const sizing = 'max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden'
  const active = 'active:opacity-50'
  return (
    <div
      className={`bg-gray-400 rounded-lg shadow-xl ${sizing} ${hover} ${transition} ${active}`}
      onClick={action}
    >
      {image && <img className="object-cover w-full h-24" src={image} alt="alt" />}
      <div className="p-3">
        <Title text={{ main: title, side: subtitle }} size="small" color="dark" />
      </div>
    </div>
  )
}

export default Chip
