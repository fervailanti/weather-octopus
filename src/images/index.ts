const req = (route: string) => require(`../images/gallery/${route}`)

export const images: any = {
  common: {
    octopus: req('common/octopus.png'),
    reload: req('common/reload.svg')
  },
  cities: {
    newYork: req('cities/new_york.jpg'),
    rome: req('cities/rome.jpg'),
    madrid: req('cities/madrid.jpg'),
    tokyo: req('cities/tokyo.jpg'),
    paris: req('cities/paris.jpg'),
    berlin: req('cities/berlin.jpg'),
  },
  weather: {
    icons: {
      sun: req('icons/sun.svg'),
      moon: req('icons/moon.svg'),
      sunCloudy: req('icons/sun-cloudy.svg'),
      moonCloudy: req('icons/moon-cloudy.svg'),
      clouds: req('icons/clouds.svg'),
      wind: req('icons/wind.svg'),
      rain: req('icons/rain.svg'),
      showerRain: req('icons/shower-rain.svg'),
      sunRain: req('icons/sun-rain.svg'),
      moonRain: req('icons/moon-rain.svg'),
      storm: req('icons/storm.svg'),
      snow: req('icons/snow.svg'),
      mist: req('icons/mist.svg'),
      forecast: req('icons/forecast.svg')
    }
  }
}