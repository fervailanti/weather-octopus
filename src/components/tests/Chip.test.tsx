import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Chip from '../Chip'
import { images } from '../../images'

describe('Chip Component', () => {
  it('Should render correctly and execute action passed by props', () => {
    const dummyAction = jest.fn()
    const chip = render(
      <Chip title="New York" subtitle="USA" image={images.cities.newYork} action={dummyAction} />
    )
    expect(chip.container).toMatchSnapshot()
    fireEvent.click(screen.getByText('New York'))
    expect(dummyAction).toHaveBeenCalledTimes(1)
  })
})
