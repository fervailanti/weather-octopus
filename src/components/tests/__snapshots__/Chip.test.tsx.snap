// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component Should render correctly and execute action passed by props 1`] = `
<div>
  <div
    class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
  >
    <img
      alt="alt"
      class="object-cover w-full h-24"
      src="new_york.jpg"
    />
    <div
      class="p-3"
    >
      <h2
        class="text-gray-900"
      >
        <span
          class="font-semibold text-xl"
        >
          New York
        </span>
        <span
          class="text-sm"
        >
          , 
          USA
        </span>
      </h2>
    </div>
  </div>
</div>
`;
